### PR TITLE
Apply some code modernization for Java 8 and recent Jenkins

### PR DESCRIPTION
**Pull request by jvz**
This updates code to use currently supported language features.

Signed-off-by: Matt Sicker <boards@gmail.com>

### CODE REVIEW

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java [hunk 1]
```diff
@@ -1016,13 +1016,8 @@ public void remove() {
...
         List<CredentialsProvider> providers =
-                new ArrayList<CredentialsProvider>(ExtensionList.lookup(CredentialsProvider.class));
-        for (Iterator<CredentialsProvider> iterator = providers.iterator(); iterator.hasNext(); ) {
-            CredentialsProvider p = iterator.next();
-            if (!p.isEnabled()) {
-                iterator.remove();
-            }
-        }
+                new ArrayList<>(ExtensionList.lookup(CredentialsProvider.class));
+        providers.removeIf(p -> !p.isEnabled());
```
Comment (jglick) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java:1020]: or

```suggestion
                ExtensionList.lookup(CredentialsProvider.class).stream().filter(CredentialsProvider::isEnabled).collect(Collectors.toList());
```

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java [hunk 2]
```diff
@@ -1035,13 +1030,8 @@ public void remove() {
...
         List<CredentialsProvider> providers =
-                new ArrayList<CredentialsProvider>(ExtensionList.lookup(CredentialsProvider.class));
-        for (Iterator<CredentialsProvider> iterator = providers.iterator(); iterator.hasNext(); ) {
-            CredentialsProvider p = iterator.next();
-            if (!p.isEnabled(context)) {
-                iterator.remove();
-            }
-        }
+                new ArrayList<>(ExtensionList.lookup(CredentialsProvider.class));
+        providers.removeIf(p -> !p.isEnabled(context));
```
Comment (jglick) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java:1034]: similarly

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java [hunk 3]
```diff
@@ -1349,14 +1339,9 @@ protected boolean _isApplicable(Descriptor<?> descriptor) {
...
-            result = new ArrayList<CredentialsDescriptor>(result);
-        }
-        for (Iterator<CredentialsDescriptor> iterator = result.iterator(); iterator.hasNext(); ) {
-            CredentialsDescriptor d = iterator.next();
-            if (!_isApplicable(d)) {
-                iterator.remove();
-            }
+            result = new ArrayList<>(result);
         }
+        result.removeIf(d -> !_isApplicable(d));
```
Comment (jglick) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java:1344]: similarly

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsSelectHelper.java [hunk 4]
```diff
@@ -335,12 +335,7 @@ public String format(Object... objects) {
                 resolverByName.put(shortName, resolverByName.containsKey(shortName) ? ContextResolver.NONE : r);
             }
         }
-        for (Iterator<ContextResolver> iterator = resolverByName.values().iterator(); iterator.hasNext(); ) {
-            ContextResolver r = iterator.next();
-            if (r == ContextResolver.NONE) {
-                iterator.remove();
-            }
-        }
+        resolverByName.values().removeIf(r -> r == ContextResolver.NONE);
```
Comment (jglick) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsSelectHelper.java:338]: or stream

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsSelectHelper.java [hunk 5]
```diff
@@ -366,12 +361,7 @@ public String format(Object... objects) {
                 providerByName.put(simpleName, providerByName.containsKey(simpleName) ? CredentialsProvider.NONE : r);
             }
         }
-        for (Iterator<CredentialsProvider> iterator = providerByName.values().iterator(); iterator.hasNext(); ) {
-            CredentialsProvider p = iterator.next();
-            if (p == CredentialsProvider.NONE) {
-                iterator.remove();
-            }
-        }
+        providerByName.values().removeIf(p -> p == CredentialsProvider.NONE);
```
Comment (jglick) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsSelectHelper.java:364]: or stream

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsStore.java [hunk 6]
```diff
@@ -434,14 +430,9 @@ protected boolean _isApplicable(Descriptor<?> descriptor) {
...
-                result = new ArrayList<CredentialsDescriptor>(result);
-            }
-            for (Iterator<CredentialsDescriptor> iterator = result.iterator(); iterator.hasNext(); ) {
-                CredentialsDescriptor d = iterator.next();
-                if (!_isApplicable(d) || !provider._isApplicable(d) || !d.isApplicable(provider)) {
-                    iterator.remove();
-                }
+                result = new ArrayList<>(result);
             }
+            result.removeIf(d -> !_isApplicable(d) || !provider._isApplicable(d) || !d.isApplicable(provider));
```
Comment (jglick) [inline src/main/java/com/cloudbees/plugins/credentials/CredentialsStore.java:435]: or stream

File: src/main/java/com/cloudbees/plugins/credentials/domains/URIRequirementBuilder.java [hunk 7]
```diff
@@ -162,12 +147,7 @@ public URIRequirementBuilder withoutHostname() {
      */
     @NonNull
     public URIRequirementBuilder withoutHostnamePort() {
-        for (Iterator<DomainRequirement> iterator = requirements.iterator(); iterator.hasNext(); ) {
-            DomainRequirement r = iterator.next();
-            if (r instanceof HostnamePortRequirement) {
-                iterator.remove();
-            }
-        }
+        requirements.removeIf(r -> r instanceof HostnamePortRequirement);
```
Comment (jglick) [inline src/main/java/com/cloudbees/plugins/credentials/domains/URIRequirementBuilder.java:150]: (Reviewer note: these cases are deliberately mutating, it seems.)

File: src/test/java/com/cloudbees/plugins/credentials/CredentialsProviderTest.java [hunk 8]
```diff
@@ -285,7 +285,7 @@ public void testNodeCredentialFingerprintsAreRemovedForNonExistentNodes() throws
         DumbSlave addedSlave = new DumbSlave("added-slave",
                 "dummy", "/home/test/slave", "1", Node.Mode.NORMAL, "remote",
                 new JNLPLauncher(),
-                RetentionStrategy.INSTANCE, Collections.<NodeProperty<?>>emptyList());
+                RetentionStrategy.INSTANCE, Collections.emptyList());
```
Comment (jglick) [inline src/test/java/com/cloudbees/plugins/credentials/CredentialsProviderTest.java:288]: FYI, this constructor is deprecated and you can more easily use a simpler version.

File: src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java (hunk 1 above)
Comment (jvz) [reply src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java:1020]: Nice catch. I'll do some manual modernizing here, too.

Comment (jglick) [reply src/main/java/com/cloudbees/plugins/credentials/CredentialsProvider.java:1020]: Sure, as time permits.

File: src/main/java/com/cloudbees/plugins/credentials/UserCredentialsProvider.java [hunk 9]
```diff
@@ -178,7 +178,7 @@ public CredentialsStore getStore(@CheckForNull ModelObject object) {
                 }
             }
         }
-        return new ArrayList<C>();
+        return new ArrayList<>();
```
Comment (res0nance) [inline src/main/java/com/cloudbees/plugins/credentials/UserCredentialsProvider.java:181]:
```suggestion
        return Collections.emptyList();
```

**Comment by jvz** [comment]
GitHub outage caused build failure. Closing to retry.

**Pull request merged**